src/UserSpaceInstrumentation/DummyProcessForTest.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#include "DummyProcessForTest.h"
+
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "absl/time/clock.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace fs = std::filesystem;
+
+namespace {
+
+// Create a file at path.
+void Touch(fs::path p) {
+  std::ofstream ofs(p);
+  ofs << "\n";
+  ofs.close();
+}
+
+}  // namespace
+
+DummyProcessForTest::DummyProcessForTest() {
+  flag_file_run_child_ = std::tmpnam(nullptr);
+  flag_file_child_started_ = std::tmpnam(nullptr);
+  Touch(flag_file_run_child_);
+  pid_ = fork();
+  CHECK(pid_ != -1);
+  // Start the workload and have the parent wait for the startup to complete.
+  if (pid_ == 0) {
+    DummyWorkload();
+    exit(0);
+  } else {
+    while (!fs::exists(flag_file_child_started_)) continue;
+  }
+}
+
+DummyProcessForTest::~DummyProcessForTest() {
+  fs::remove(flag_file_child_started_);
+  fs::remove(flag_file_run_child_);
+  int status;
+  waitpid(pid_, &status, 0);
+  CHECK(WIFEXITED(status));
+}
+
+void DummyProcessForTest::DummyWorker() {
+  constexpr auto kTimeToLiveMs = std::chrono::milliseconds(15);
+  const auto deadline = std::chrono::system_clock::now() + kTimeToLiveMs;
+  while (true) {
+    if (std::chrono::system_clock::now() > deadline) {
+      break;
+    }
+  }
+  std::lock_guard<std::mutex> lock(joinable_mutex_);
+  joinable_threads_.emplace(std::this_thread::get_id());
+}
+
+void DummyProcessForTest::DummyWorkload() {
+  size_t kNumThreads = 4;
+  std::vector<std::thread> threads;
+  while (fs::exists(flag_file_run_child_) || !threads.empty()) {
+    // Spawn as many threads as there are missing.
+    while (threads.size() < kNumThreads && fs::exists(flag_file_run_child_)) {
+      threads.emplace_back(std::thread(&DummyProcessForTest::DummyWorker, this));
+    }
+    Touch(flag_file_child_started_);
```

Comment:
So this file is "touched" every time a new child is created, yes?

Reply:
Yes, it would be enought to do it in the first round. But why bother?